gradle/documentation/documentation.gradle

```diff
@@ -20,34 +20,41 @@ configure(rootProject) {
   def refguideUrlVersion = project.baseVersion.replaceFirst(/^(\d+)\.(\d+).*$/, '$1_$2')
 
   ext {
-    luceneDocUrl = project.propertyOrDefault('lucene.javadoc.url', {
-      if (project.version != project.baseVersion) {
-        // non-release build
-        new File(project('lucene:documentation').buildDir, 'site').toURI().toASCIIString().minus(~'/$')
-      } else {
-        // release build
-        "https://lucene.apache.org/core/${urlVersion}"
-      }
-    }())
-
-    solrDocUrl = project.propertyOrDefault('solr.javadoc.url', {
-      if (project.version != project.baseVersion) {
-        // non-release build
-        new File(project('solr:documentation').buildDir, 'site').toURI().toASCIIString().minus(~'/$')
-      } else {
-        // release build
-        "https://lucene.apache.org/solr/${urlVersion}"
-      }
-    }())
+    if (!skipLucene) {
+      luceneDocUrl = project.propertyOrDefault('lucene.javadoc.url', {
+        if (project.version != project.baseVersion) {
+          // non-release build
+          new File(project('lucene:documentation').buildDir, 'site').toURI().toASCIIString().minus(~'/$')
+        } else {
+          // release build
+          "https://lucene.apache.org/core/${urlVersion}"
+        }
+      }())
+    }
 
-    solrRefguideUrl = project.propertyOrDefault('solr.refguide.url', "https://lucene.apache.org/solr/guide/${refguideUrlVersion}")
+    // SOLR ONLY
+    if (!skipSolr) {
+      solrDocUrl = project.propertyOrDefault('solr.javadoc.url', {
+        if (project.version != project.baseVersion) {
+          // non-release build
+          new File(project('solr:documentation').buildDir, 'site').toURI().toASCIIString().minus(~'/$')
+        } else {
+          // release build
+          "https://lucene.apache.org/solr/${urlVersion}"
+        }
+      }())
+
+      solrRefguideUrl = project.propertyOrDefault('solr.refguide.url', "https://lucene.apache.org/solr/guide/${refguideUrlVersion}")
+    }
   }
 
   task documentation() {
     group = 'documentation'
     description = 'Generate all documentation'
 
-    dependsOn ':lucene:documentation:assemble'
+    if (!skipLucene) {
+      dependsOn ':lucene:documentation:assemble'
+    }
     dependsOn ':solr:documentation:assemble'
```

Comment:
check skipSolr here?

Reply:
For some reason this actually works...

Reply:
top-level documentation wasn't part of assemble, that's why it worked.